android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -349,6 +350,18 @@ constructor(
       return (INCLUSIVE_SIX_DIGIT_MINIMUM..INCLUSIVE_SIX_DIGIT_MAXIMUM).random()
     }
 
+    fun filterList(
+      list: List<Resource>,
+      value: String,
+      fhirPathExpression: String
+    ): List<Resource> {
+      return list.filter {
+        (fhirPathDataExtractor.extractData(it, "$fhirPathExpression = '$value'").first() as
+            BooleanType)
+          .value
+      }
+    }
+
     fun filterList(list: List<Resource>, attribute: String, attributeValue: Any) =
```

Comment:
will remove (**_filterList & get Value method_**) this once new logic is reviewed.